Mathlib/Algebra/Polynomial/Basic.lean

```diff
@@ -1303,10 +1303,10 @@ protected instance repr [Repr R] [DecidableEq R] : Repr R[X] :=
       (p.support.sort (· ≤ ·));
     match termPrecAndReprs with
     | [] => "0"
-    | [(tprec, t)] => if prec ≥ tprec then Lean.Format.paren t else t
+    | [(tprec, t)] => if tprec ≤ prec then Lean.Format.paren t else t
```

Comment:
You're already fixing that in #12933. Maybe you can just merge the PRs into one?

Reply:
Fair enough; I have cherry-picked these changes into the other PR. I will re-use this one for more changes (the whole branch has about +- 300 lines of fixes).